kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/Service2Service.java

```diff
@@ -46,22 +47,42 @@ public TaskExchange addToken(TaskExchange taskExchange) {
 		return taskExchange;
 	}
 
+	/**
+	 * Add a token to a UserExchange message for sending.
+	 *
+	 * @param userExchange
+	 * @return
+	 */
+	public UserExchange addToken(UserExchange userExchange) {
+		userExchange.setToken(userToken.getToken());
+		return userExchange;
+	}
+
 	/**
 	 * Initialise the RequestScope.
 	 *
 	 * @param taskExchange The TaskExchange object
 	 */
 	public void initialiseScope(TaskExchange taskExchange) {
-		log.debug(taskExchange);
 		scope.init(jsonb.toJson(taskExchange));
 		logToken();
 	}
 
+	/**
+	 * Initialise the RequestScope.
+	 *
+	 * @param userExchange
+	 */
+	public void initialiseScope(UserExchange userExchange) {
+		scope.init(jsonb.toJson(userExchange));
+		logToken();
+	}
+
 	/**
 	 * log token
 	 */
 	public void logToken() {
-		log.infof("USER [%s] : [%s]", userToken.getUserCode(), userToken.getUsername());
+		log.debugf("USER [%s] : [%s]", userToken.getUserCode(), userToken.getUsername());
```

Comment:
If we're seriously worried about performance, we need to be careful about how many formatted strings we have kicking around the system. I know its not always a consideration, but it is something good to keep in the back of our minds if we decide to chase performance later

https://stackoverflow.com/questions/513600/should-i-use-javas-string-format-if-performance-is-important